File: Baekjoon/S5/C#/S5_수들의 합.cs
https://www.acmicpc.net/problem/1789

using System;

namespace Beakjoon
{
    class Program
    {
        static void Main(string[] args)
        {
            ulong n = ulong.Parse(Console.ReadLine());
            ulong num = 1;
            ulong sum = 0;

            while (true)
            {
                sum += num;

                if (sum > n)
                    break;

                num++;
            }

            Console.Write(--num);
        }
    }
}

File: Baekjoon/S5/JAVA/S5_수들의 합.java
//https://www.acmicpc.net/problem/1789

import java.io.*;

public class Main{
    public static void main(String args[]) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        long n = Long.parseLong(br.readLine());
        
        long cnt = 0;
        while(true){
            cnt++;
            if(cnt*(cnt+1)/2 > n)
                break;
        }
        
        System.out.print(cnt-1);
        br.close();
    }
}